test/test_document_review_checker.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+from unittest.mock import patch, MagicMock
+import tempfile
+import os
+
+from bin import document_review_checker as check
+
+
+class TestDocumentReviewChecker(unittest.TestCase):
+    def setUp(self) -> None:
+        self.file_path = tempfile.mkdtemp()
+        with tempfile.NamedTemporaryFile(
+                dir=self.file_path, delete=False, suffix=".erb.md") as self.file:
+            self.file.write(b"last_reviewed_on: 2020-03-14")
+
+    def tearDown(self) -> None:
+        self.addCleanup(os.remove, self.file.name)
+
+    def test_documents_due_for_review(self):
+        documents = check.get_documents_due_for_review(self.file_path)
+        self.assertIn(self.file.name, documents)
+
+    def test_no_documents_to_review(self):
+        new_file_path = tempfile.mkdtemp()
+        documents = check.get_documents_due_for_review(new_file_path)
+        self.assertNotIn(self.file.name, documents)
+
+
+class TestFixingDocumentDates(unittest.TestCase):
+    def setUp(self) -> None:
+        self.fix = True
+        self.file_path = tempfile.mkdtemp()
+
+    def test_with_file_to_fix(self):
+        with tempfile.NamedTemporaryFile(
+                dir=self.file_path, delete=False, suffix=".erb.md") as file:
+            file.write(b"last_reviewed_on: 2020-03-14")
+
+        self.addCleanup(os.remove, file.name)
+
+        documents = check.get_documents_due_for_review(self.file_path)
+        self.assertIn(file.name, documents)
+
+        check.fix_date(file.name)
+
+        documents = check.get_documents_due_for_review(self.file_path)
+        self.assertNotIn(file.name, documents)
+
+
+class TestMainFunction(unittest.TestCase):
```

Comment:
Would it be better to treat the `main` function unit test as an integration test? I.e. run through a happy path scenario for the full call chains? 🤔